cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricProcessorService.java

```diff
@@ -0,0 +1,216 @@
+import io.cdap.cdap.spi.data.transaction.TxCallable;
+import io.cdap.cdap.store.DefaultNamespaceStore;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * AppFabric Processor Service which runs messaging subscriber services.
+ */
+public class AppFabricProcessorService extends AbstractIdleService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AppFabricProcessorService.class);
+
+  private final ProgramRuntimeService programRuntimeService;
+  private final ApplicationLifecycleService applicationLifecycleService;
+  private final ProgramNotificationSubscriberService programNotificationSubscriberService;
+  private final ProgramStopSubscriberService programStopSubscriberService;
+  private final AuditLogSubscriberService auditLogSubscriberService;
+  private final RunRecordCorrectorService runRecordCorrectorService;
+  private final RunDataTimeToLiveService runDataTimeToLiveService;
+  private final ProgramRunStatusMonitorService programRunStatusMonitorService;
+  private final RunRecordMonitorService runRecordCounterService;
+  private final CoreSchedulerService coreSchedulerService;
+  private final CredentialProviderService credentialProviderService;
+  private final NamespaceCredentialProviderService namespaceCredentialProviderService;
+  private final ProvisioningService provisioningService;
+  private final BootstrapService bootstrapService;
+  private final SystemAppManagementService systemAppManagementService;
+  private final SourceControlOperationRunner sourceControlOperationRunner;
+  private final RepositoryCleanupService repositoryCleanupService;
+  private final OperationNotificationSubscriberService operationNotificationSubscriberService;
+  private final ScheduleNotificationSubscriberService scheduleNotificationSubscriberService;
+  private final CConfiguration cConf;
+  private final TransactionRunner transactionRunner;
+  private MetricsCollectionService metricsCollectionService;
+
+  /**
+   * Construct the AppFabricProcessorService with service factory and cConf coming from guice injection.
+   */
+  @Inject
+  public AppFabricProcessorService(CConfiguration cConf,
+      @Nullable MetricsCollectionService metricsCollectionService,
+      ProgramRuntimeService programRuntimeService,
+      RunRecordCorrectorService runRecordCorrectorService,
+      ProgramRunStatusMonitorService programRunStatusMonitorService,
+      ApplicationLifecycleService applicationLifecycleService,
+      ProgramNotificationSubscriberService programNotificationSubscriberService,
+      ProgramStopSubscriberService programStopSubscriberService,
+      AuditLogSubscriberService auditLogSubscriberService,
+      CoreSchedulerService coreSchedulerService,
+      CredentialProviderService credentialProviderService,
+      NamespaceCredentialProviderService namespaceCredentialProviderService,
```

Comment:
why do we want to move `CredentialProviderService`?

Reply:
Not needed. Removed.